todo_app/app.py

```diff
@@ -1,34 +1,75 @@
 from flask.wrappers import Request
-from flask import Flask, render_template, redirect, url_for, request
-import backend
+from flask import Flask, render_template, redirect, url_for, request, jsonify
 
+import backend
+from flask_dance.contrib.github import github
+from flask_login import logout_user, login_required
 
 app = Flask(__name__)
 
+from models import db, login_manager
+from oauth import github_blueprint
+
+app.secret_key = 'notsosecretkey'
+app.register_blueprint(github_blueprint, url_prefix="/login")
+
+db.init_app(app)
+login_manager.init_app(app)
+
+with app.app_context():
+    db.create_all()
+
 @app.route("/",methods = ['POST', 'GET'])
 def index():
-    if request.method == 'POST':
+    if not github.authorized:
+        return redirect(url_for("github.login"))
+    
+    elif request.method == 'POST':
+        resp = github.get("/user")
+        assert resp.ok
```

Comment:
This checks the user has authenticated, but not what their rights are, which means that I can bypass your authorization checks currently since they only occur on the frontend; see the task [I've added to your Heroku app](http://stormy-tundra-71419.herokuapp.com/)! We should certainly add some checks here to ensure the user has the right permissions; we could look at pulling that check out into a decorator if we split the GET & POST functions out here or could explicitly check per function